mock/reads_store.go

```diff
@@ -0,0 +1,53 @@
+package mock
```

Comment:
Hand-rolled this mock to maximize the ability to re-use code from IDPE in the provider tests.